src/Vision/grippyCode.py

```diff
@@ -0,0 +1,168 @@
+import gripV2
+import cv2
+import numpy as np
+import threading
+import math
+from networktables import NetworkTables
+
+
+cond = threading.Condition()
+notified = [False]
+
+def connectionListener(connected, info):
+    print(info, '; Connected=%s' % connected)
+    with cond:
+        notified[0] = True
+        cond.notify()
+
+NetworkTables.initialize(server='10.1.66.2')
+NetworkTables.addConnectionListener(connectionListener, immediateNotify=True)
+table = NetworkTables.getTable('Vision Correction Table')
+with cond:
+    if not notified[0]:
+        cond.wait()
+
+cap = cv2.VideoCapture(1)
+cap.set(10,30)
+
+goalFinder = gripV2.GripiPipeline()
+
+rightAngle = -15
+leftAngle = -75
+deadzone = 15
+
+
+def findPairOffset(pair):
+    leftcX = pair[0][0][0]
+    rightcX = pair[1][0][0]
+    pairMidpoint = (leftcX + rightcX) / 2
+    pairOffset = abs(imageMidpoint - pairMidpoint)
```

Comment:
Where does `imageMidpoint` come from?

Reply:
image midpoint is defined in normalizeImage fuction on line 91

Reply:
That's a local variable, that only exists inside `normalizeImage`. You shouldn't be able to access it here.

Please don't leak variables all over the place, it makes kittens sad. 😿

Reply:
No more sad Kittens! 😿 !=